src/TestFramework/TestFramework/Assertions/Assert.AreEqual.cs

```diff
@@ -10,6 +12,256 @@ namespace Microsoft.VisualStudio.TestTools.UnitTesting;
 /// </summary>
 public sealed partial class Assert
 {
+    [InterpolatedStringHandler]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public readonly struct AssertAreEqualInterpolatedStringHandler<TArgument>
+    {
+        private readonly StringBuilder? _builder;
+        private readonly object? _expected;
+        private readonly object? _actual;
+
+        public AssertAreEqualInterpolatedStringHandler(int literalLength, int formattedCount, TArgument? expected, TArgument? actual, out bool shouldAppend)
+        {
+            _expected = expected!;
+            shouldAppend = AreEqualFailing(expected, actual);
+            if (shouldAppend)
+            {
+                _builder = new StringBuilder(literalLength + formattedCount);
+                _expected = expected;
+                _actual = actual;
+            }
+        }
+
+        public AssertAreEqualInterpolatedStringHandler(int literalLength, int formattedCount, TArgument? expected, TArgument? actual, IEqualityComparer<TArgument>? comparer, out bool shouldAppend)
+        {
+            shouldAppend = AreEqualFailing(expected, actual, comparer);
+            if (shouldAppend)
+            {
+                _builder = new StringBuilder(literalLength + formattedCount);
+                _expected = expected;
+                _actual = actual;
+            }
+        }
+
+        public AssertAreEqualInterpolatedStringHandler(int literalLength, int formattedCount, IEquatable<TArgument>? expected, IEquatable<TArgument>? actual, out bool shouldAppend)
+        {
+            shouldAppend = AreEqualFailing(expected, actual);
+            if (shouldAppend)
+            {
+                _builder = new StringBuilder(literalLength + formattedCount);
+                _expected = expected;
+                _actual = actual;
+            }
+        }
+
+        internal void FailIfNeeded()
+        {
+            if (_builder is not null)
+            {
+                FailAreEqual(_expected, _actual, _builder.ToString());
+            }
+        }
+
+        public readonly void AppendLiteral(string value) => _builder!.Append(value);
+
+        public readonly void AppendFormatted<T>(T value) => _builder!.Append(value);
+
+#if NETCOREAPP3_1_OR_GREATER
+        public readonly void AppendFormatted(ReadOnlySpan<char> value) => _builder!.Append(value.ToString());
```

Comment:
Why calling `ToString` when the `Append` has an overload with `ReadOnlySpan<char>`?